xunit.runner.wpf/ViewModel/MainViewModel.cs

```diff
@@ -476,13 +476,37 @@ private List<Task> RunTests()
             }
         }
 
+        private List<TraitViewModel> traitsToAdd = new List<TraitViewModel>();
+        private List<TestCaseViewModel> testCasesToAdd = new List<TestCaseViewModel>();
+        private Task updateTestCasesTask = null;
+
         private void OnTestDiscovered(TestCaseData testCaseData)
         {
+            var scheduler = TaskScheduler.FromCurrentSynchronizationContext();
             var traitList = testCaseData.TraitMap.Count == 0
                 ? ImmutableArray<TraitViewModel>.Empty
                 : testCaseData.TraitMap.SelectMany(pair => pair.Value.Select(value => new TraitViewModel(pair.Key, value))).ToImmutableArray();
-            this.allTestCases.Add(new TestCaseViewModel(testCaseData.SerializedForm, testCaseData.DisplayName, testCaseData.AssemblyPath, traitList));
-            this.traitCollectionView.Add(traitList);
+
+            traitsToAdd.AddRange(traitList);
+            testCasesToAdd.Add(new TestCaseViewModel(testCaseData.DisplayName, testCaseData.AssemblyPath, traitList));
+
+            if (updateTestCasesTask != null)
+            {
+                return;
+            }
+
+            updateTestCasesTask = Task.Delay(TimeSpan.FromSeconds(0.5)).ContinueWith(UpdateUI, scheduler);
```

Comment:
Why not just update the timer in https://github.com/Pilchie/xunit.runner.wpf/blob/master/xunit.runner.wpf/Impl/RemoteTestUtil.BackgroundRunner.cs to fire messages more sparingly? The callback could be called with a list of test cases rather than once for each test case.